api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -872,21 +872,34 @@ type AgentNodePoolPlatform struct {
 	AgentLabelSelector *metav1.LabelSelector `json:"agentLabelSelector,omitempty"`
 }
 
+type AzureVMImageType int
+
+const (
+	ImageID AzureVMImageType = iota
+	AzureMarketplace
+)
+
+// AzureVMImage represents the different types of image data that can be provided for an Azure VM
+type AzureVMImage struct {
+	Type                 AzureVMImageType  `json:"azureImageType"`
+	ImageID              *AzureImageInfo   `json:"imageID,omitempty"`
```

Comment:
what happens if you specify both? do we need a union discriminator? also this needs to be documented